legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/domain/DomainParseTreeWalker.java

```diff
@@ -395,6 +397,10 @@ private org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain
         func.returnType = ctx.functionTypeSignature().type().getText();
         func.returnMultiplicity = this.buildMultiplicity(ctx.functionTypeSignature().multiplicity().multiplicityArgument());
         func.sourceInformation = this.walkerSourceInformation.getSourceInformation(ctx);
+        String functionSignature =   LazyIterate.collect(func.parameters, HelperValueSpecificationGrammarComposer::getParameterSignature).select(Objects::nonNull).makeString("__")
+            + "__" + HelperValueSpecificationGrammarComposer.getClassSignature(func.returnType) + "_" + HelperValueSpecificationGrammarComposer.getMultiplicitySignature(func.returnMultiplicity) + "_";
+        String terseSignatureSuffix =  func.parameters.size() > 0 ? "_" + functionSignature : functionSignature;
+        func.name = PureGrammarParserUtility.fromIdentifier(ctx.qualifiedName().identifier()) + terseSignatureSuffix;
```

Comment:
lets move this logic out and call it like `buildFullFunctionSignature` to make it clear we are assigning the full function name to the .name property.

Reply:
Dont we also have already defined methods we could use here like `getFunctionSignature` and getFunctionName ?